### PR TITLE
Specify summarization, destruction, and options

### PR DESCRIPTION
**Pull request by domenic**
This is most of the remaining normative text for the summarizer API.

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/writing-assistance-apis/pull/18.html" title="Last updated on Nov 29, 2024, 2:20 AM UTC (a02c611)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/writing-assistance-apis/18/da6e057...a02c611.html" title="Last updated on Nov 29, 2024, 2:20 AM UTC (a02c611)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -521,10 +531,387 @@ The <dfn attribute for="AISummarizer">format</dfn> getter steps are to return [=
...
+      <th>Meaning
+  <tbody>
+    <tr>
+      <th>"<dfn enum-value for="AISummarizerType">tl;dr</dfn>"
+      <td>
+        <p>The summary should be short and to the point, providing a quick overview of the input, suitable for a busy reader.
+    <tr>
+      <th>"<dfn enum-value for="AISummarizerType">teaser</dfn>"
+      <td>
+        <p>The summary should focus on the most interesting or intriguing parts of the input, designed to draw the the reader in to read more.
```
Comment (tomayac) [inline index.bs:798]:
```suggestion
        <p>The summary should focus on the most interesting or intriguing parts of the input, designed to draw the reader in to read more.
```

File: index.bs [hunk 2]
```diff
@@ -521,10 +531,387 @@ The <dfn attribute for="AISummarizer">format</dfn> getter steps are to return [=
...
+    <tr>
+      <th>"<dfn enum-value for="AISummarizerType">key-points</dfn>"
+      <td>
+        <p>The summary should extract the most important points from the input, presented as a bulleted list.
+    <tr>
+      <th>"<dfn enum-value for="AISummarizerType">headline</dfn>"
+      <td>
+        <p>The summary should effectively containing the main point of the input in a single sentence, in the format of an article headline.
```
Comment (tomayac) [inline index.bs:806]:
```suggestion
        <p>The summary should effectively contain the main point of the input in a single sentence, in the format of an article headline.
```

File: index.bs [hunk 3]
```diff
@@ -521,10 +531,387 @@ The <dfn attribute for="AISummarizer">format</dfn> getter steps are to return [=
...
+  <tbody>
+    <tr>
+      <th>"<dfn enum-value for="AISummarizerLength">short</dfn>"
+      <td>
+        <p>The guidance is dependent on the value of {{AISummarizerType}}:
+
+        <dl class="switch">
+          : "{{AISummarizerType/tl;dr}}"
+          : "{{AISummarizerType/teaser}}"
+          :: The summary should fit within a one sentence.
```
Comment (tomayac) [inline index.bs:824]:
```suggestion
          :: The summary should fit within one sentence.
```

File: index.bs [hunk 4]
```diff
@@ -521,10 +531,387 @@ The <dfn attribute for="AISummarizer">format</dfn> getter steps are to return [=
...
+        </dl>
+    <tr>
+      <th>"<dfn enum-value for="AISummarizerLength">medium</dfn>"
+      <td>
+        <p>The guidance is dependent on the value of {{AISummarizerType}}:
+
+        <dl class="switch">
+          : "{{AISummarizerType/tl;dr}}"
+          : "{{AISummarizerType/teaser}}"
+          :: The summary should fit within a one short paragraph.
```
Comment (tomayac) [inline index.bs:838]:
```suggestion
          :: The summary should fit within 1 short paragraph.
```

File: index.bs [hunk 5]
```diff
@@ -521,10 +531,387 @@ The <dfn attribute for="AISummarizer">format</dfn> getter steps are to return [=
...
+        </dl>
+    <tr>
+      <th>"<dfn enum-value for="AISummarizerLength">long</dfn>"
+      <td>
+        <p>The guidance is dependent on the value of {{AISummarizerType}}:
+
+        <dl class="switch">
+          : "{{AISummarizerType/tl;dr}}"
+          : "{{AISummarizerType/teaser}}"
+          :: The summary should fit within a one paragraph.
```
Comment (tomayac) [inline index.bs:852]:
```suggestion
          :: The summary should fit within 1 paragraph.
```

File: index.bs [hunk 6]
```diff
@@ -521,10 +531,389 @@ The <dfn attribute for="AISummarizer">format</dfn> getter steps are to return [=
...
+
+When summarization fails, the following possible reasons may be surfaced to the web developer. This table lists the possible {{DOMException}} [=DOMException/names=] and the cases in which an implementation should use them:
+
+<table class="data">
+  <thead>
+    <tr>
+      <th>{{DOMException}} [=DOMException/name=]
+      <th>Scenarios
+  <tbody>
+    <tr>
```
[the remark this answers is not in the document]
Comment (domenic) [reply index.bs:896]: How can the backend session be destroyed? Is it via the destroy() method or the AbortSignal? Then it should be an AbortError. Or is there some other possibility?

[the remark this answers is not in the document]
Comment (gjc13) [reply index.bs:896]: It is caused by the backend aborting itself. Maybe we'd like to have another error type for backend failures or simply use `UnknownError` instead?

[the remark this answers is not in the document]
Comment (domenic) [reply index.bs:896]: Do you know why the backend would abort itself? Like, how would I trigger that as a user or developer?

But yes, I think UnknownError might be best for such cases.

**Pull request merged**